openstack/containerorchestration/v1/common/errors.go

```diff
@@ -0,0 +1,118 @@
+package common
+
+import (
+	"encoding/json"
+	"github.com/gophercloud/gophercloud"
+	"strings"
+)
+
+// ErrorResponse represents failed response
+type ErrorResponse struct {
+	gophercloud.ErrUnexpectedResponseCode
+	Errors []Error `json:"errors" required:"true"`
+}
+
+// Error represents a Magnum API error message
+type Error struct {
+	Status    int    `json:"status"`
+	Code      string `json:"code"`
+	Title     string `json:"title"`
+	Detail    string `json:"detail"`
+	RequestID string `json:"request_id,omitempty"`
+}
+
+// Error returns the error message details from the server
+func (e *ErrorResponse) Error() string {
+	errors := []string{}
+	for _, e := range e.Errors {
+		errors = append(errors, e.Detail)
+	}
+	return strings.Join(errors, "\n")
+}
+
+func (e *ErrorResponse) unwrapError(r gophercloud.ErrUnexpectedResponseCode) bool {
```

Comment:
I wasn't returning the error because the error handling doesn't rely upon the content of the error at all. Basically I was following the cast pattern, which returns the cast value and a boolean representing if the cast was successful.

How about naming it `tryUnwrapError` and the calling it like this: `if e.tryUnwrapError(r) {...}`